### DISCUSSION TITLE
Remove spree_users.ship_address_id and bill_address_id columns

### DISCUSSION THREAD

**Discussion by spaghetticode** [solidusio/solidus#4573] [New Features or Ideas]
Since the introduction of the user's address book, we've been relying on those fields on the user model less and less, to the point that nowadays we have just a couple of points in `core` where they're used and a few specs that verify those fields are kept in sync only in the address book specs.

These fields don't impact the user's ship and bill address, as these are now `has_one :through` associations. Extracted from the [address book](https://github.com/MeUndies/solidith/blob/master/app/models/hobbes/order/address_book.rb):

```ruby
      has_many :addresses, through: :user_addresses

      has_one :default_user_bill_address, ->{ default_billing }, class_name: 'Spree::UserAddress', foreign_key: 'user_id'
      has_one :bill_address, through: :default_user_bill_address, source: :address

      has_one :default_user_ship_address, ->{ default_shipping }, class_name: 'Spree::UserAddress', foreign_key: 'user_id'
      has_one :ship_address, through: :default_user_ship_address, source: :address
```

The areas where we still reference those fields are:

* `Spree::UserAddressBook#persist_order_address` 
   * https://github.com/solidusio/solidus/blob/master/core/app/models/concerns/spree/user_address_book.rb#L81
   * https://github.com/solidusio/solidus/blob/master/core/app/models/concerns/spree/user_address_book.rb#L90
* `Spree::UserAddressBook#persist_order_address`
  * https://github.com/solidusio/solidus/blob/master/core/app/models/concerns/spree/user_address_book.rb#L171
  * https://github.com/solidusio/solidus/blob/master/core/app/models/concerns/spree/user_address_book.rb#L172
   
If they're actually so useless nowadays, I'm thinking it may be time to consider at least completely removing the remnants of the old functionality in the code (i.e. stop keeping those fields synced), and possibly entirely dropping the columns in a DB migration in order to clear some confusion.

**Comment by vitorderobe** [comment]
Sounds good to me! It's nice to have a single source of truth for `ship_address` and `bill_address` to avoid confusion.

**Comment by waiting-for-dev** [comment]
It makes total sense
